test/integration/productsController.test.js

```diff
@@ -0,0 +1,366 @@
+/* eslint-disable no-unused-expressions */
+const chai = require('chai')
+const { expect } = chai
+const chaiHttp = require('chai-http')
+
+const app = require('../../server')
+const productsModel = require('../../models/products')
+const userModel = require('../../models/users')
+const firestore = require('../../utils/firestore')
+const authService = require('../../services/authService')
+const cookieName = config.get('userToken.cookieName')
+
+chai.use(chaiHttp)
+
+describe('Products', function () {
+  let jwt
+  let createdUserId
+  before(async function () {
+    const user = {
+      first_name: 'Prakash',
+      last_name: 'C',
+      yoe: 0,
+      img: './img.png',
+      github_id: 'prakashchoudhary07',
+      username: 'prakash'
+    }
+    // Adding user
+    const { userId } = await userModel.addOrUpdate(user)
+    createdUserId = userId
+    // Generatinf JWT
+    jwt = authService.generateAuthToken({ userId })
+  })
+  describe('GET /products', function () {
+    afterEach(async function () {
+      const product = {
+        id: 'coffee',
+        image: 'coffeeImage',
+        name: 'coffee',
+        manufacturer: 'RDS',
+        price: 55,
+        category: 'food',
+        usage: ['To make coffee']
+      }
+      await productsModel.addProduct(product)
+    })
+    it('should return empty ', function (done) {
+      chai
+        .request(app)
+        .get('/products')
+        .end((err, res) => {
+          if (err) {
+            return done()
+          }
+          expect(res).to.have.status(200)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.contain.keys('message', 'products')
+          expect(res.body.message).to.equal('No products found')
+          expect(res.body.products).to.be.a('object').that.is.empty
+          return done()
+        })
+    })
+    it('Should return product details', function (done) {
+      chai
+        .request(app)
+        .get('/products')
+        .end((err, res) => {
+          if (err) {
+            return done()
+          }
+          expect(res).to.have.status(200)
+          expect(res.body).to.be.a('object')
+          expect(res.body).to.contain.keys('message', 'products')
+          expect(res.body.message).to.equal('Products returned successfully!')
+          expect(res.body.products).to.be.a('object')
+          expect(res.body.products.coffee).to.have.all.keys(
+            'id',
+            'image',
+            'name',
+            'manufacturer',
+            'price',
+            'category',
+            'usage'
```

Comment:
Better to have a list of these keys and using the spread operator here to check for the keys?